File: kits/ts/simple/lux/Cell.ts
import {Position} from "./Position";
import {CityTile} from "./CityTile";

export interface Resource {
  type: string;
  amount: number;
}

export class Cell {
  public pos: Position;
  public resource: Resource = null;
  public citytile: CityTile = null;
  public road = 0;

  public constructor(x: number, y: number) {
    this.pos = new Position(x, y);
  }

  public hasResource(): boolean {
    return this.resource !== null && this.resource.amount > 0;
  }
}

File: kits/java/simple/lux/Cell.java
package lux;

public class Cell {
  public Position pos;
  public Resource resource = null;
  public double road = 0;
  public CityTile citytile = null;
  public Cell(int x, int y) {
    this.pos = new Position(x, y);
  }
  public boolean hasResource() {
    return this.resource != null && this.resource.amount > 0;
  }
}
